diff --git a/kernel/Kbuild b/kernel/Kbuild
index 1e79d0b..c587f7e 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -25,14 +25,21 @@
 #
 # $(1): The object files.
 # $(2): The CFLAGS to add for those object files.
 #
 
+ifeq ($(shell [ $(VERSION) -gt 5 ] || [ $(VERSION) -eq 5 -a $(PATCHLEVEL) -gt 3 ]; echo $$? ),0)
+ASSIGN_PER_OBJ_CFLAGS = \
+ $(foreach _cflags_variable, \
+ $(addprefix CFLAGS_,$(1)), \
+ $(eval $(_cflags_variable) += $(2)))
+else
 ASSIGN_PER_OBJ_CFLAGS = \
  $(foreach _cflags_variable, \
  $(addprefix CFLAGS_,$(notdir $(1))), \
  $(eval $(_cflags_variable) += $(2)))
+endif
 
 
 #
 # Include the specifics of the individual NVIDIA kernel modules.
 #
diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index c00c826..40c54f6 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -3583,11 +3583,11 @@ case "$5" in
         fi
 
         TAB='	'
 
         if [ -f "$OUTPUT/Module.symvers" ] && \
-             grep -e "^[^${TAB}]*${TAB}[^${TAB}]*${TAB}vmlinux" \
+             grep -e "^[^${TAB}]*${TAB}[^${TAB}]*.*${TAB}vmlinux" \
                      "$OUTPUT/Module.symvers" >/dev/null 2>&1; then
             exit 0
         fi
 
         echo "The Module.symvers file is missing, or does not contain any"
diff --git a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
index bc31de7..7eefe95 100644
--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -38,10 +38,11 @@
 #if defined(NV_DRM_AVAILABLE)
 
 #include "nvidia-drm-ioctl.h"
 
 #include <drm/drmP.h>
+#include <linux/version.h>
 
 /*
  * Commit fcd70cd36b9b ("drm: Split out drm_probe_helper.h")
  * moves a number of helper function definitions from
  * drm/drm_crtc_helper.h to a new drm_probe_helper.h.
@@ -657,11 +658,15 @@ static const struct drm_ioctl_desc nv_drm_ioctls[] = {
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 };
 
 static struct drm_driver nv_drm_driver = {
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0)
     .driver_features        = DRIVER_GEM | DRIVER_PRIME | DRIVER_RENDER,
+#else
+    .driver_features        = DRIVER_GEM | DRIVER_RENDER,
+#endif
 
     .gem_free_object        = nv_drm_gem_free,
 
     .ioctls                 = nv_drm_ioctls,
     .num_ioctls             = ARRAY_SIZE(nv_drm_ioctls),
-- 
2.27.0

